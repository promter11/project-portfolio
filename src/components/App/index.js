import React, { useState, useEffect } from "react";

import Preloader from "../Preloader";
import Sidebar from "../../containers/Sidebar";
import Content from "../../containers/Content";

import styles from "./App.module.scss";

const App = () => {
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const timer = setTimeout(
      () => setLoading((value) => !value),
      (Math.random() * (3 - 1) + 1) * 1000
    );

    return () => clearTimeout(timer);
  }, []);

  return (
    <div className={styles.app}>
      <Preloader loading={loading} />
      <Sidebar />
      <Content />
    </div>
  );
};

export default App;
