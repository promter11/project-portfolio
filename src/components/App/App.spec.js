import React from "react";
import { shallow } from "enzyme";

import App from "./index";

jest.useFakeTimers();

describe("App component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls a function with setTimeout", (done) => {
    jest.spyOn(React, "useEffect").mockImplementationOnce((f) => f());
    jest.advanceTimersByTime(3000);

    setTimeout(() => {
      expect(wrapper.find("Preloader")).toBeFalsy();
    }, 3000);

    jest.useRealTimers();

    done();
  });
});
