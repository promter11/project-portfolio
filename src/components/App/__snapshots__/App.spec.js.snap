// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="app"
>
  <Preloader
    loading={true}
  />
  <Connect(Sidebar) />
  <Connect(Content) />
</div>
`;
